### PR TITLE
add in batching for spot creation

### PR DESCRIPTION
**Pull request by alyssa-blair**
Added text box that takes spot numbers as an input in a format similar to 1,5-7,9. This will allow users to batch add parking spots that have the same information. Also added support for choosing spot availability times using a days of the week buttons and a range picker for the time. The steps of this range picker are 15 minutes for now but we can update this later if needed.

image
image
image

### CODE REVIEW

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt [hunk 1]
```diff
@@ -135,122 +119,109 @@ fun AddSpotView(onBackClick: () -> Unit, onSpotAdded: () -> Unit, addSpotViewMod
...
+                    ) {
+                        Column {
+                            Text("Building Type", fontSize = 12.sp, color = Color.Gray)
+                            Text(selectedBuildingType)
+                        }
+
+                        Spacer(modifier = Modifier.weight(1f))
+
+                        IconToggleButton(
+                            checked = buildingTypeExpanded,
```
Comment (wrighted) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt:175]: Is there not a builtin dropdown menu?

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt [hunk 2]
```diff
@@ -310,12 +281,32 @@ fun AddSpotView(onBackClick: () -> Unit, onSpotAdded: () -> Unit, addSpotViewMod
 
             Button(
                 onClick = {
-                    val address = searchedSuggestion.value?.name ?: "dev spot"
-                    val spot = Spot("999", "dev3", address, currentLocation.longitude, currentLocation.latitude)
-                    scope.launch {
-                        addSpotViewModel.createSpot(spot)
+                    val createdSpotNumbers = parseNumberRange(spotNumbers)
+
+                    if (selectedBuildingType == "House") {
```
Comment (wrighted) [inline composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt:286]: Mine is showing the `dev spot x`, why is the searchSuggestion `null` sometimes?

I also think this if condition is unnecessary, if `parseNumberRange(spotNumbers)` can just return the right list of spots (including the single case) then we can reduce logic here.

**Review by wrighted — APPROVED** [review]
Few things but seems to mostly work.

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt (hunk 1 above)
Comment (alyssa-blair) [reply composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt:175]: I did try to use that but it wasn't simple to format the list to resemble our other dropdown menus. I could try to mess around with it but might not be worth it for now.

Comment (wrighted) [reply composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt:175]: In that case, maybe making the whole field clickable would make it better.

Comment (alyssa-blair) [reply composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt]: I updated it to be a clickable field now

File: composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt (hunk 2 above)
Comment (alyssa-blair) [reply composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt]: I updated this to keep the spot number list as part of the UI regardless of the type of building in the latest change. I also fixed the issue with the search suggestion so it correctly propogates the selected address with the spot number.

**Review by wrighted — APPROVED** [review]
I trust you.

**Pull request merged**